### ISSUE TITLE
Create Projects contract

### ISSUE THREAD

**Issue by LNow** [labs3/daoOS#23]
**Is your feature request related to a problem? Please describe.**
In order to allow DAO members to raise funds for their projects we need to have Clarity contract that with following functionalities:
- create-project
- add-milestone
- remove-milestone
- edit-milestone
- back-project/support-project
- transfer-milestones-funds

**Describe the solution you'd like**
Features listed above are bare minimum and should be our starting point. Right now we only need an outline of a working contract that we can expand in the future by adding to it things like access control, voting, refunds, time-based functions etc.

**Comment by cryptopanter** [comment]
So this is going to be the main contract ( in DAO Contract container) apart from other contract plugins?

**Comment by pxydn** [comment]
> So this is going to be the main contract ( in DAO Contract container) apart from other contract plugins?

yes, by the looks of it

**Comment by LNow** [comment]
> So this is going to be the main contract ( in DAO Contract container) apart from other contract plugins?

Yes, but it doesn't mean we can't convert it to plugin, or create it as a plugin now and convert to standard contract later.
No matter if we create it as a plugin or not, core business logic of this contract will remain the same.

**Comment by cryptopanter** [comment]
> **Is your feature request related to a problem? Please describe.**
> In order to allow DAO members to raise funds for their projects we need to have Clarity contract that with following functionalities:
> 
> * create-project
> * add-milestone
> * remove-milestone
> * edit-milestone
> * back-project/support-project
> * transfer-milestones-funds
> 
> **Describe the solution you'd like**
> Features listed above are bare minimum and should be our starting point. Right now we only need an outline of a working contract that we can expand in the future by adding to it things like access control, voting, refunds, time-based functions etc.

So can we consider it as the server's action (or system action)?... Since @falkonprods has already derived the users' action here: https://github.com/syvita/daoOS/wiki/Design-Structure:-User-Actions

**Comment by falkonprods** [comment]
Remember this isn't needed until M3...great we are working on it, but all this will come in the next  milestone...for now should just be members and funding...
